lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -224,6 +224,18 @@ public function exists(Closure $p) : bool;
      */
     public function filter(Closure $p) : self;
 
+    /**
+     * Return the value of first element found in array of elements that satisfies the $func.
+     *
+     * @param Closure $func The predicate used to find element.
+     *
+     * @return mixed|null The element if found or NULL if not.
+     *
+     * @psalm-param Closure(T=, TKey=):bool $func
+     * @psalm-return T|null
+     */
+    public function find(Closure $func);
```

Comment:
This is a BC break, so we can't add this method in 1.x. We also don't have a strategy in place on how to add new methods to the collection interface without creating a hard BC break, which is why we've previously rejected all pull requests that expand the interface. Due to custom collection implementations (e.g. in ODM) we also can't just add the methods to some of these collection classes and "assume" they'll be there.